polkadot/xcm/xcm-simulator/fuzzer/src/fuzz.rs

```diff
@@ -155,6 +158,34 @@ fn run_input(xcm_messages: [XcmMessage; 5]) {
 	println!();
 
 	for xcm_message in xcm_messages {
+		fn matches_blocklisted_messages(message: Instruction<()>) -> bool {
+			matches!(message, Transact { .. })
+		}
+		// We check XCM messages recursively for blocklisted messages
+		fn matches_recursive(message: &mut Instruction<()>) -> Vec<Instruction<()>> {
+			match message {
+				DepositReserveAsset { xcm, .. } |
+				ExportMessage { xcm, .. } |
+				InitiateReserveWithdraw { xcm, .. } |
+				InitiateTeleport { xcm, .. } |
+				TransferReserveAsset { xcm, .. } |
+				SetErrorHandler(xcm) |
+				SetAppendix(xcm) => Vec::from(xcm.inner()).iter_mut().flat_map(matches_recursive).collect(),
+				_ => vec![message.clone()],
```

Comment:
Hm, this function is called `matches_recursive`, but it dies not call the `matches_blocklisted_messages`?  
To me it looks like it unpacks them.

Reply:
Should be addressed